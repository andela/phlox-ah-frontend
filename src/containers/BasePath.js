export * from '../requests/LoginRequests';
export * from '../util/AsyncUtil';
export * from '../actionTypes/UserConstants';
export * from '../actions/MsgInfoActions';
export * from '../components/Input/Input';
export * from '../components/Button/Button';
export * from '../requests/SignupRequests';
export * from '../requests/ArticleRequests';
export * from '../requests/ProfileRequest';

