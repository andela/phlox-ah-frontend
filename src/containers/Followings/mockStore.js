export const followerStore = {
  loading: false,
  success: false,
  followUser: {
    followings: [{
      id: 1,
      username: 'johndoe',
      email: 'jd@something.com',
      Profile: {
        firstName: 'John',
        lastName: 'Doe',
        profileImage: 'https://res.cloudinary.com/sammykeyz/image/upload/v1542924983/phlox/1542924981230_me.jpg',
        gender: 'male',
        contact: 'Test street',
        bio: 'Lowkey bad guy'
      }
    }],
    followers: [{
      id: 1,
      username: 'johndoe',
      email: 'jd@something.com',
      Profile: {
        firstNa: 'John',
        lastName: 'Doe',
        profileImage: 'https://res.cloudinary.com/sammykeyz/image/upload/v1542924983/phlox/1542924981230_me.jpg',
        gender: 'male',
        contact: 'Test street',
        bio: 'Lowkey bad guy'
      }
    }],
  }
};
