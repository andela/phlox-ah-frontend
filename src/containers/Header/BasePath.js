export * from '../../actionTypes/UserConstants';
export * from '../../util/AsyncUtil';
