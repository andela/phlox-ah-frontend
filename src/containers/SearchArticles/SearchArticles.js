import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Row, Col, Button } from 'react-materialize';

import { ArticleCard } from '../../components/ArticleCard/ArticleCard';
import Sidebar from '../Sidebar/Sidebar';
import { searchArticles } from '../../requests/SearchRequests';
import './SearchArticles.scss';


/**
 * @class ViewAllArticles
 * @extends {Component}
 */
class SearchArticles extends Component {
  /**
   *Creates an instance of Home.
   * @param {object} props
   * @memberof ViewAllArticles
   */
  constructor(props) {
    super();
    this.initialState = { searchBy: 'article', query: '' };

    this.state = { ...this.initialState };

    this.change = this.change.bind(this);
    this.submit = this.submit.bind(this);
    this.swithSearchBy = this.swithSearchBy.bind(this);
  }

  /**
   * @returns {func} tag
   */
  componentDidMount() {
    const { search } = this.props.location;
    if (search) {
      this.props.searchArticles({ query: search, searchBy: 'article' });
    }
  }

  /**
   * @memberOf handleChangeEvent
   * @method handleChangeEvent
   * @param {*} e
   * @return {*} setstate
   */
  change(e) {
    this.setState({
      query: e.target.value
    });
  }

  /**
   * @description - This method changes the searchBy filter
   * @param {objecj} e
   * @returns {object} null
   * @memberof SearchArticles
   */
  swithSearchBy(e) {
    const searchBy = e.target.value;
    const searchQuery = `?${searchBy}=${this.state.query}`;
    this.props.searchArticles({ query: searchQuery, searchBy });
    this.setState({
      searchBy
    });
  }

  /**
   * @description - This method runs when search form is submitted
   * @param {objecj} e
   * @returns {object} null
   * @memberof SearchArticles
   */
  submit(e) {
    e.preventDefault();
    const { searchBy, query } = this.state;
    const searchQuery = `?${searchBy}=${query}`;
    this.props.searchArticles({ query: searchQuery, searchBy });
  }

  /**
   * @description - This method shows all articles
   * @returns {jsx} - jsx
   */
  showSearchedArticles() {
    const { searchResult } = this.props;
    return searchResult.map((article) => {
      const username = article.User ? article.User.username : '';
      return (
        <Col s={12} m={12} l={12} xl={6} key={article.id}>
          <ArticleCard
            size="medium"
            pic={article.imgUrl}
            title={article.title}
            description={article.description}
            createdAt={article.createdAt}
            author={username}
            slug={article.slug}
          />
        </Col>);
    });
  }

  /**
   *
   * @description - This method renders the jsx for this component
   * @returns {jsx} - jsx
   */
  render() {
    const {
      success, failure, loading, error
    } = this.props;
    const { searchBy, query } = this.state;

    return (
      <div>
        <main>
          <div className="site-container ">
          <div className="my-container">
            <Row>
              <Col s={12}>
                <Row>
                  <Row>
                    <Col s={12} className="column-2">
                    <form onSubmit={this.submit}>
                      <input
                        type="text"
                        onChange={this.change}
                        placeholder="Search authors heaven"
                        name={'query'}
                        value={query}
                        className="searchInput"
                        id="searchInput"
                      />
                      <Row className="filter-row">
                        <Col s={2}>
                          <h6 className="text-bold">Search By: </h6>
                        </Col>
                        <Col s={10}>
                          <input
                            className={searchBy === 'author' ? 'active' : ''}
                            type="button" value={'author'} onClick={this.swithSearchBy}
                            name="myname"
                          /><span className="divider">|</span>
                          <input
                            className={searchBy === 'article' ? 'active' : ''}
                            type="button"
                            value={'article'}
                            onClick={this.swithSearchBy}
                            name="myname"
                          /><span className="divider">|</span>
                          <input
                            className={searchBy === 'tag' ? 'active' : ''}
                            type="button"
                            value={'tag'}
                            onClick={this.swithSearchBy}
                            name="myname"
                          />
                        </Col>
                      </Row>
                    </form>
                    </Col>
                  </Row>
                  <div className="mt-5">
                    {success && this.showSearchedArticles() }
                    { failure && <h5 className="center">{ error }</h5>}
                    { loading && <h5 className="center">loading...</h5>}
                  </div>
                </Row>
              </Col>
            </Row>
            </div>
          </div>
        </main>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  articles: state.article.articles,
  loading: state.search.loading,
  success: state.search.success,
  failure: state.search.failure,
  error: state.search.error,
  searchResult: state.search.searchResult
});

SearchArticles.propTypes = {
  searchResult: PropTypes.array,
  searchArticles: PropTypes.func,
  match: PropTypes.object,
  loading: PropTypes.bool,
  success: PropTypes.bool,
  failure: PropTypes.bool,
  error: PropTypes.string,
  location: PropTypes.shape({
    search: PropTypes.string
  })
};

export default connect(mapStateToProps, { searchArticles })(SearchArticles);
