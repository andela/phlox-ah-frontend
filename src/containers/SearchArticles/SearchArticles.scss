@import '../../styles/variables';
.column-2 {
  padding-right: 35px;
  width: 65%;
  .divider{
    margin-left:20px;
    margin-right:20px;
    color:#C4C4C4;
  }
  #searchInput{
    border: 1px solid #ADA9A9;
    padding-left:20px;
    border-radius:5px;
  }
  input[type=button]{
      width: 100px;
      padding: 10px 20px;
      display: inline-block;
      border: 1px solid #ccc;
      box-sizing: border-box;
      border-radius:5px;
      font-weight:bold;
  }
  .active{
    background-color: rgb(116, 112, 112);
  }
  .searchInput{
    margin-bottom:30px;
  }
  .row .col {
    padding-right: 0.9rem;
  }
}
.preloaderDiv {
  position: relative;
  height: 280px;
  margin-bottom: 10px;
  background: #E5E5E5;
}
h5.capitalize {
  margin-top: 0px;
}
h5.no-tags {
  color: #6F6A6A;

}
.my-container{
  min-height: 0px;
  padding: 0px 50px;
}
.filter-row{
  margin-top: 30px;
  h6{
    color:#6F6A6A;
    font-size: 20px;
  }
}
.text-bold{
  font-weight:bold;
}
.mt-5{
  margin-top: 70px;
}
@media #{$tablet} {
  .my-container{
    min-height: 0px;
    padding: 0px 0px;
  }
}
@media #{$mobile} {
  padding-right: 10px;
  .my-container{
    min-height: 0px;
    padding: 0px 0px;
  }
}
