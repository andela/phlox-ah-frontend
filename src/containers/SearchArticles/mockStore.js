export const articleStore = {
  search: {
    loading: true,
    success: true,
    failure: false,
    searchResult: [
      {
        id: 3,
        title: 'Second article by nelso',
        body: 'The second article by Nelson v Second article by Nelson.The second article by Nelson.<p><br></p>',
        slug: 'Second-article-by-nelson-09afb545-4110-4656-8378-c6b6f192eb92',
        description: 'Description of second article by nelson',
        imgUrl: 'https://res.cloudinary.com/dafcdw02e/image/upload/v1542892294/phlox/1542892286288_adult-agent-approval-684385.jpg',
        readTime: 1,
        ratingAverage: 0,
        ArticlesTags: {
          articleId: 3,
          tagId: 1,
          createdAt: '2018-11-22T13:11:20.097Z',
          updatedAt: '2018-11-22T13:11:20.097Z'
        }
      }
    ],
    error: null
  }
};
