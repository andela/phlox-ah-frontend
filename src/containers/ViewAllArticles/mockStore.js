export const articleStore = {
  article: {
    loading: true,
    success: true,
    failure: false,
    articles: [
      {
        id: 3,
        title: 'Second article by nelson',
        body: 'The second article by Nelson v Second article by Nelson.The second article by Nelson.<p><br></p>',
        slug: 'Second-article-by-nelson-09afb545-4110-4656-8378-c6b6f192eb92',
        description: 'Description of second article by nelson',
        imgUrl: 'https://res.cloudinary.com/dafcdw02e/image/upload/v1542892294/phlox/1542892286288_adult-agent-approval-684385.jpg',
        readTime: 1,
        userId: 2,
        categoryId: 1,
        ratingAverage: 0,
        featured: 0,
        status: 'published',
        createdAt: '2018-11-22T13:11:20.033Z',
        updatedAt: '2018-11-22T13:11:44.537Z',
        Tags: [
          {
            id: 1,
            name: 'fake news',
            createdAt: '2018-11-22T13:11:19.925Z',
            updatedAt: '2018-11-22T13:11:19.925Z',
            ArticlesTags: {
              articleId: 3,
              tagId: 1,
              createdAt: '2018-11-22T13:11:20.097Z',
              updatedAt: '2018-11-22T13:11:20.097Z'
            }
          }
        ],
        Category: {
          id: 1,
          category: 'Technology',
          createdAt: '2018-11-22T23:00:00.000Z',
          updatedAt: '2018-11-22T23:00:00.000Z'
        },
        User: {
          username: 'nelson',
          Profile: null
        },
        likes: [
          {
            id: 3,
            like: null,
            createdAt: '2018-11-22T13:12:23.783Z',
            updatedAt: '2018-11-22T13:17:11.857Z',
            articleSlug: 'Second-article-by-nelson-09afb545-4110-4656-8378-c6b6f192eb92',
            userId: 2,
            User: {
              username: 'nelson',
              email: 'nelson.nnaji@andela.com'
            }
          }
        ]
      },
      {
        id: 2,
        title: 'new test article by udochukwu',
        body: '<p>This enables you to deliver on time regardless of unforeseen blockers. Products are built on timelines. Sometimes these timelines are based on launch dates, financial projections, product roadmaps, strategic objectives .etc. Every task done and feature implemented contributes to meeting the timelines set for the overall product. In addition, features have dependencies and timelines on one feature contribute to timelines on others. If you don’t learn how to estimate timelines for delivery, it will translate to you making promises you can’t keep. Risks you didn’t foresee will always creep up and mess up your delivery timelines. Mastering the ability to estimate accurately enables you to build a great relationship with your clients /partners and to reinforce your brand as a professional developer who can be trusted.</p>\n\n<p>Every task you pick up on a team project will require you to scope it out, assign it a weight based on its complexity and estimate how long it will take you to complete it. Every bug you encounter will be a potential blocker to meeting your estimated timeline. Every new concept or technology that you need to learn on the path to implementing a feature is a potential hindrance to meeting estimated timelines. Software development is problem-solving on the clock. If you fail to master the ability to foresee and account for risks that arise in the process of building a product, you will struggle to estimate properly and constantly fail to deliver on time. You will basically be a dev who ‘talks a lot, doesn’t deliver!’</p>',
        slug: 'new-test-article-by-udochukwu-af9329ec-e692-4fe9-8af8-d684dab47178',
        description: 'Description here and here too',
        imgUrl: 'https://images.pexels.com/photos/179912/pexels-photo-179912.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
        readTime: 1,
        userId: 1,
        categoryId: 1,
        ratingAverage: 4,
        featured: 0,
        status: 'published',
        createdAt: '2018-11-20T08:00:03.743Z',
        updatedAt: '2018-11-21T14:07:53.239Z',
        Tags: [],
        Category: {
          id: 1,
          category: 'Technology',
          createdAt: '2018-11-22T23:00:00.000Z',
          updatedAt: '2018-11-22T23:00:00.000Z'
        },
        User: {
          username: 'nnaji',
          Profile: null
        },
        likes: [
          {
            id: 2,
            like: false,
            createdAt: '2018-11-20T11:23:02.778Z',
            updatedAt: '2018-11-22T12:31:07.855Z',
            articleSlug: 'new-test-article-by-udochukwu-af9329ec-e692-4fe9-8af8-d684dab47178',
            userId: 2,
            User: {
              username: 'nelson',
              email: 'nelson.nnaji@andela.com'
            }
          },
          {
            id: 1,
            like: true,
            createdAt: '2018-11-20T10:47:12.913Z',
            updatedAt: '2018-11-20T11:13:16.155Z',
            articleSlug: 'new-test-article-by-udochukwu-af9329ec-e692-4fe9-8af8-d684dab47178',
            userId: 1,
            User: {
              username: 'nnaji',
              email: 'nnaji_udochukwu@yahoo.com'
            }
          }
        ]
      }
    ]
  },
  comments: {
    comment: [
      {
        id: 3,
        comment: 'you be goat oo',
        createdAt: '2018-11-21T14:00:11.810Z',
        updatedAt: '2018-11-21T14:00:11.810Z',
        articleSlug: 'new-test-article-by-udochukwu-af9329ec-e692-4fe9-8af8-d684dab47178',
        userId: 2,
        User: {
          username: 'nelson',
          email: 'nelson.nnaji@andela.com',
          Profile: null
        },
        likes: [],
        Replies: []
      }
    ]
  },
  user: {
    isAuth: true,
    loading: false
  },
  params: {
    articleslug: 'article-slxug'
  }
};
