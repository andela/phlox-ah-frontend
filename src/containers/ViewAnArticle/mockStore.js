export const articleStore = {
  Articles: {
    loading: false,
    success: false,
    Article: {
      loading: true,
      success: false,
      failure: false,
      article: {
        title: 'new test article by udochukwu',
        body: 'Body here',
        description: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. ',
        imgUrl: 'https://images.pexels.com/photos/788485/pexels-photo-788485.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
        readTime: 1,
        createdAt: '2018-11-12T18:16:18.630Z',
        updatedAt: '2018-11-12T18:16:18.630Z',
        User: {
          username: 'nnaji',
          Profile: {
            profileImage: 'https://images.pexels.com/photos/1065084/pexels-photo-1065084.jpeg?cs=srgb&dl=beautiful-beauty-brown-eyes-1065084.jpg&fm=jpg'
          }
        },
        Tags: [
          {
            id: 1,
            name: 'tech',
            createdAt: '2018-09-22T23:00:00.000Z',
            updatedAt: '2018-09-22T23:00:00.000Z',
          },
          {
            id: 2,
            name: 'animals',
            createdAt: '2018-09-22T23:00:00.000Z',
            updatedAt: '2018-09-22T23:00:00.000Z',
          }
        ],
        likes: [],
      }
    }
  },
  params: {
    articleslug: 'article-slxug'
  },
};
