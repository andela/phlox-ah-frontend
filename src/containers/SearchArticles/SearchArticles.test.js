import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import SearchArticles from './SearchArticles';
import { articleStore } from './mockStore';
import { SEARCH_ARTICLES } from '../../actionTypes';
import { asyncActions } from '../BasePath';


const mockStore = configureMockStore([thunk]);
const store = mockStore(articleStore);

const props = {
  location: {
    search: ''
  }
};

let component;
let myComponent;

describe('<SearchArticles/>', () => {
  test('renders the ArticlePage Container', () => {
    component = shallow(
      <Provider store={store}>
        <SearchArticles { ...props } />
      </Provider>
    );
    myComponent = component.dive({ context: { store } }).dive();
    expect(component.exists()).toBe(true);
  });
  it('should render without throwing an error', () => {
    expect(component).toMatchSnapshot();
  });
  it('should have a main container', () => {
    expect(myComponent.find('.site-container').exists()).toBe(true);
  });
  it('should have a search input', () => {
    expect(myComponent.find('.searchInput').exists()).toBe(true);
  });
  it('should have a filter row', () => {
    expect(myComponent.find('.filter-row').exists()).toBe(true);
  });
  it('should have an input with class name active', () => {
    expect(myComponent.find('input.active').exists()).toBe(true);
  });
  it('should have function that displays the search results', () => {
    expect(myComponent.instance().showSearchedArticles).toBeDefined();
  });
  it('should have function that changes the search filter', () => {
    expect(myComponent.instance().swithSearchBy).toBeDefined();
  });
  it('should have a function  that handles input change event', () => {
    expect(myComponent.instance().change).toBeDefined();
  });
  it('should have articles state as array', () => {
    expect(myComponent.instance().state.searchBy).toBeDefined();
  });
  it('should create an action to search for articles', () => {
    const payload = true;
    const expectedAction = {
      type: 'SEARCH_ARTICLES_SUCCESS',
      payload
    };
    expect(asyncActions(SEARCH_ARTICLES).success(payload)).toEqual(expectedAction);
  });
});
