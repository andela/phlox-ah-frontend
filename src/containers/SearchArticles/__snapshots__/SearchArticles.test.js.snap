// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchArticles/> should render without throwing an error 1`] = `
<Connect(SearchArticles)
  location={
    Object {
      "search": "",
    }
  }
/>
`;
